### PR TITLE
Add more device helper tests

### PR DESCRIPTION
**Pull request by DaQuirm**
More TDD goodness :smile_cat: 
@larsblumberg @ghophp

### CODE REVIEW

File: public/app/common/helpers/device/test/device-helpers.spec.js [hunk 1]
```diff
@@ -1,17 +1,74 @@
...
-        actions: [],
-        components: [ { properties: { type: 'nest', capabilities: [ { } ] } } ],
-        links: [{ rel: ['self'] }]
-      };
-      var device = DeviceHelpers.augmentDevice(nestWebservice);
-      expect(device.hidden).toBeTruthy();
+      var nestService = DeviceHelpers.augmentDevice(nestWebservice);
+      expect(nestService.hidden).toEqual(true);
+      var augmentedDevice = DeviceHelpers.augmentDevice(device);
+      expect(augmentedDevice.hidden).toEqual(false);
```
Comment (larsblumberg) [inline public/app/common/helpers/device/test/device-helpers.spec.js]: The second `expect` should be another test case named _'does not flag normal devices as hidden'_

File: public/app/common/helpers/device/test/device-helpers.spec.js [hunk 2]
```diff
@@ -1,17 +1,74 @@
...
+    it('returns full device url', function () {
+      var url = DeviceHelpers.getDeviceUrl(device);
+      expect(url).toEqual(device.links[0].href);
+    });
+  });
+
+  describe('isHiddenDevice', function () {
+    it('returns true if device has the string "nest acccount" in its name property', function () {
+      var isHidden = DeviceHelpers.isHiddenDevice(device);
+      expect(isHidden).toEqual(false);
```
Comment (larsblumberg) [inline public/app/common/helpers/device/test/device-helpers.spec.js]: same as above, should be _'returns false device is a usual device'_

**Comment by larsblumberg** [comment]
Apart from the comments, LGTM

**Comment by DaQuirm** [comment]
Travis build is failing because of an [issue](https://github.com/cjohansen/Sinon.JS/issues/761) with Sinon

File: public/app/common/helpers/device/test/device-helpers.mock.coffee [hunk 3]
```diff
@@ -0,0 +1,153 @@
+module.exports =
+
+  device:
+    entities: [
+      {
```
Comment (larsblumberg) [inline public/app/common/helpers/device/test/device-helpers.mock.coffee]: I see you still using curly brackets in coffescript!

**Comment by larsblumberg** [comment]
LGTM

PS: Still curly brackets in coffeescript? oO

**Comment by DaQuirm** [comment]
Yeah, you still need them to avoid ambiguity :pouting_cat: 
But no more `,` and `;` :smile_cat:

**Pull request merged**